library/libraries.js

```diff
@@ -1753,5 +1753,17 @@ var d41d8cd98f00b204e9800998ecf8427e_LibraryDetectorTests = {
 
             return false;
         }
-    }
+    },
+    'Joomla': {
+        id: 'joomla',
+        icon: 'joomla',
+        url: 'https://www.joomla.org/',
+        npm: null,
+        test: function (win) {
+            const generatorMeta = document.querySelector('meta[name=Generator][content^="Joomla"]');
+            const version = generatorMeta ? generatorMeta.getAttribute("content").replace(/\D+/gi,'') : UNKNOWN_VERSION;
+
+            return false;
```

Comment:
This is a great start, thanks again for getting this in ❤️

So apparently, the `meta` generator tag can be removed on any Joomla site. I think we may be missing a lot of detections if we only rely on this, and I also was trying to find if we can get version numbers from `/language/en-GB/en-GB.xml` since that seems to be more fool-proof.

What I came up with:

```
test: async function (win) { 
  const generatorMeta = !!document.querySelector('meta[name="generator"][content*="Joomla"]');
  const joomlaComponent = !!document.querySelector('[href*="index.php?option=com_"]');

  try {
      const response = await fetch('/language/en-GB/en-GB.xml');
      const manifest = await response.text();
      const parser = new DOMParser();
      
      let parsedManifest = null;
      if (manifest) {
          parsedManifest = parser.parseFromString(manifest, "text/xml");
      }

      if (parsedManifest && !!parsedManifest.getElementsByTagName("version").length) {
          return  { version: parsedManifest.getElementsByTagName("version")[0].childNodes[0].nodeValue }
      } else if (generatorMeta || joomlaComponent) {
          return { version: UNKNOWN_VERSION };
      }
  }
  catch (err) {
      return false;
  }

  return false;
}
```

What do you think? This works when I test it on a number of Joomla sites, but I would love an extra sanity check with your experience to make sure I'm not missing anything :)

Reply:
You are absolutely right!
I added the code you suggested and will have double checked by other Joomla contributors!
Thanks @housseindjirdeh

Reply:
The check here is pretty general. You could also check for the existence of /administrator/manifests/libraries/joomla.xml